resources/resource-extensbility/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,432 @@
+
+    1. Deb lists the resources types he has in his environment
+
+        ```bash
+        rad resource-provider list -e myenv
+        ```
+        Radius lists the custom resource types that Deb has registered in his environment
+
+2. Amy is a system integrator who is familiar with OpenAPI schema format and wants to author a custom resource PostgreSQL in Radius
+
+    1. Amy uses the radCLI to scaffold a yaml template definition
+
+        ```bash
+        rad resource-provider init plaid --template-kind yaml
+        ```
+        A sample yaml file is created in the application folder.
+
+    1. Amy authors the schema for plaid in typespec format
+        
+        ```
+        namespace: 'Applications.Datastores'
+        apiVersion: 'v1'
+        scope: '/planes/radius/Applications.Datastores'
+        resourceTypes:
+            'Applications.Datastores/postgreSQL':
+                name: 'Name of the resource'
+                description: 'PostgreSQL' 
+                properties: 
+                required: ['host', 'env', 'secrets']
+                host:
+                    type: 'string'
+                    description: 'Hostname'
+                env:
+                    type: 'string'
+                    description: 'Radius environment'
+                app:
+                    type: 'string'
+                    description: 'Radius application'
+                secrets:
+                    connectionstring:
+                    type: 'string'
+                    description: 'Connection string for PostgreSQL'
+        ```
+
+    1. Amy adds a version for the schema
+
+         ```
+        namespace: 'Applications.Datastores'
+        apiVersion: 'v1'
+        scope: '/planes/radius/Applications.Datastores'
+        resourceTypes:
+            'Applications.Datastores/postgreSQL':
+                name: 'Name of the resource'
+                description: 'PostgreSQL' 
+                properties: 
+                required: ['host', 'env', 'secrets']
+                host:
+                    type: 'string'
+                    description: 'Hostname'
+                env:
+                    type: 'string'
+                    description: 'Radius environment'
+                app:
+                    type: 'string'
+                    description: 'Radius application'
+                secrets:
+                    connectionstring:
+                    type: 'string'
+                    description: 'Connection string for PostgreSQL'
+        ```
+   1. Amy registers the postgreSQL schema to an existing provider in Radius
+    
+        ```bash
+        rad resource-type add -rp Applications --template-kind yaml -e myenv -v v1
+        ```
+        
+        Radius complies the schema and registers the custom resource type Plaid in UCP
+        
+        Question : Will Applications continue to be a resource provider to accept new resource types ? 
```

Comment:
I don't totally get the question, so I'm guessing. I think you might be asking about the use of the `Applications` namespace. 

I think we'd tell users that `Applications.Xyz` is reserved for the official Radius project. That doesn't stop users from breaking the rules, but it does make it clear what the expectations are.

Reply:
Yeah the question is around the use of existing namespace `Applications` and will we continue to accept new types  like postgreSQL?

Reply:
So are you asking what our process would be for accepting a contribution like that? I think that's an important topic. Want to take a stab at it?

Reply:
During the design review, we agreed to move the open-sourcing and contributions experience to a separate document.